src/Tests/Microsoft.Diagnostics.Monitoring.HostingStartup.UnitTests/ParameterCapturing/Pipeline/ParameterCapturingPipelineTests.cs

```diff
@@ -0,0 +1,292 @@
+                });
+
+            TaskCompletionSource<Guid> onStopCallbackSource = new(TaskCreationOptions.RunContinuationsAsynchronously);
+            TestParameterCapturingCallbacks callbacks = new(
+                onCapturingStop: (requestId) =>
+                {
+                    onStopCallbackSource.TrySetResult(requestId);
+                });
+
+            ParameterCapturingPipeline pipeline = new(probeManager, callbacks);
+            StartCapturingParametersPayload payload = CreateStartCapturingPayload(Timeout.InfiniteTimeSpan);
+
+            using CancellationTokenSource cts = new();
+            Task pipelineTask = pipeline.RunAsync(cts.Token);
+            pipeline.SubmitRequest(payload);
+
+            // Act
+            pipeline.RequestStop(payload.RequestId);
+
+            // Assert
+            await probeManagerStopSource.Task;
+            Guid stoppedRequest = await onStopCallbackSource.Task;
+            Assert.Equal(payload.RequestId, stoppedRequest);
+        }
+
+        [Fact]
+        public async Task Request_StopsAfterDuration()
+        {
+            // Arrange
+            TestFunctionProbesManager probeManager = new();
+
+            TaskCompletionSource<Guid> onStopCallbackSource = new(TaskCreationOptions.RunContinuationsAsynchronously);
+            TestParameterCapturingCallbacks callbacks = new(
+                onCapturingStop: (requestId) =>
+                {
+                    onStopCallbackSource.TrySetResult(requestId);
+                });
+
+            ParameterCapturingPipeline pipeline = new(probeManager, callbacks);
+            StartCapturingParametersPayload payload = CreateStartCapturingPayload(TimeSpan.FromSeconds(1));
+
+            using CancellationTokenSource cts = new();
+            Task pipelineTask = pipeline.RunAsync(cts.Token);
+
+            // Act
+            pipeline.SubmitRequest(payload);
+
+            // Assert
+            Guid stoppedRequest = await onStopCallbackSource.Task;
+            Assert.Equal(payload.RequestId, stoppedRequest);
+        }
+
+        [Fact]
+        public async Task RunAsync_ThrowsOnCapturingStopFailure()
+        {
+            // Arrange
+            Exception thrownException = new Exception("test");
+            TestFunctionProbesManager probeManager = new(
+                onStop: () =>
+                {
+                    throw thrownException;
+                });
+
+            TestParameterCapturingCallbacks callbacks = new();
+
+            ParameterCapturingPipeline pipeline = new(probeManager, callbacks);
+            StartCapturingParametersPayload payload = CreateStartCapturingPayload(TimeSpan.FromSeconds(1));
+
+            using CancellationTokenSource cts = new();
+
+            // Act
+            Task pipelineTask = pipeline.RunAsync(cts.Token);
+            pipeline.SubmitRequest(payload);
+
+            // Assert
+            try
+            {
+                await pipelineTask;
```

Comment:
```suggestion
               await Assert.ThrowsAsync<Exception>(pipelineTask);
```
or something to that effect?

Reply:
I think that also returns the Exception instance, so you can do:

```
Exception ex = await Assert.ThrowsAsync(...);
Assert.Equals(throwException, ex);
```